cmd/docker/docker.go

```diff
@@ -352,7 +353,7 @@ func runDocker(ctx context.Context, dockerCli *command.DockerCli) error {
 	// We've parsed global args already, so reset args to those
 	// which remain.
 	cmd.SetArgs(args)
-	err = cmd.Execute()
+	err = cmd.ExecuteContext(ctx)
```

Comment:
take note, instead of passing `ctx` to the root command through convoluted function calls I've cleaned it up a bit by instead passing context to cobra when we call `ExecuteContext`. This _should_ only affect the CLI and not any plugins that are run through the CLI.

Reply:
Ah, yes, this was one of the things to look into!

(Still catching up on latest changes in this PR, but yes, this was one where my approach was probably not right)